src/main/scala/com/snowflake/snowpark/internal/OpenTelemetry.scala

```diff
@@ -118,58 +113,48 @@ object OpenTelemetry extends Logging {
       }
     }
   }
+}
+trait SpanInfo {
+  val className: String
+  val funcName: String
+  val fileName: String
+  val lineNumber: Int
 
-  def emit(info: SpanInfo): Unit =
-    emit(info.className, info.funcName) { span =>
-      {
-        span.setAttribute("code.filepath", info.fileName)
-        span.setAttribute("code.lineno", info.lineNumber)
-        info match {
-          case ActionInfo(_, _, _, _, methodChain) =>
-            span.setAttribute("method.chain", methodChain)
-          case UdfInfo(_, _, _, _, execName, execHandler, execFilePath) =>
-            span.setAttribute("snow.executable.name", execName)
-            span.setAttribute("snow.executable.handler", execHandler)
-            span.setAttribute("snow.executable.filepath", execFilePath)
-        }
-      }
-    }
-
-  def reportError(className: String, funcName: String, error: Throwable): Unit =
-    emit(className, funcName) { span =>
-      {
-        span.setStatus(StatusCode.ERROR, error.getMessage)
-        span.recordException(error)
-      }
-    }
+  lazy private val span =
+    GlobalOpenTelemetry
+      .getTracer(s"snow.snowpark.$className")
+      .spanBuilder(funcName)
+      .startSpan()
 
-  private def emit(className: String, funcName: String)(report: Span => Unit): Unit = {
-    val name = s"snow.snowpark.$className"
-    val tracer = GlobalOpenTelemetry.getTracer(name)
-    val span = tracer.spanBuilder(funcName).startSpan()
+  private def emit[T](thunk: => T): T = {
```

Comment:
How is this method invoked? I didn't find any reference to it

Reply:
Both `emitSpan` and `reportError` invoke this function, the syntax is `emit { ... }` which equal to ` emit(...)` in Scala.

Reply:
Since I merged the reportError to the emit function, now, emit is the public API.